### ISSUE TITLE
No way to specify a dependency like `>= 1.0 && < 3.0`

### ISSUE THREAD

**Issue by kballard** [Carthage/Carthage#1688]
It looks like Carthage has no way to have a dependency that covers a custom range of versions. This is rather unfortunate. I have 2 libraries, one of which depends on the other, and can accept any 1.x or 2.x version of the other. But there's no way for this library to express this dependency, meaning I either have to just use >= 1.0 and risk breaking when the other library hits 3.0, or I have to use ~> 2.0 and force any clients of my library to update the other dependency even if that would break them.

I can see 3 ways of expressing this kind of version relationship:

1. Add a new operator `..<` like Swift's operator, so I can say `github "foo" 1.0..<3.0`
2. Add a `<` operator and a `&&` union operator so I can say `github "foo" >= 1.0 && < 3.0`
3. Add a `<` operator and let me specify the dependency twice, so I can say
  
    ```
    github "foo" >= 1.0
    github "foo" < 3.0
    ```

**Comment by mpdifran** [comment]
You wouldn't have this issue if you used semantic versioning. You should only update the major version when there's a breaking change to your library. The fact that your one library can accept version 1.x _and_ 2.x means you increased the major version without making a breaking change. If you kept your one library on 1.x, you could still use `~> 1.0`. Thus, the proposed change doesn't make sense.

You can read more about semantic versioning here: http://semver.org

**Comment by kballard** [comment]
This is completely incorrect. I did make a breaking change. It just doesn't break all clients (only one API changed in a backwards-incompatible way) and my dependent library doesn't use that API. The problem with Carthage is now my dependent library has to pick either 1.x or 2.x of the dependency even though it doesn't care and that decision should be up to the app that uses both libraries.

**Comment by mpdifran** [comment]
Ah I see. My apologies for assuming you hadn't made a breaking change. The range syntax might make sense in this case then. Or, it's not completely outrageous to force people to update both your libraries at once (i.e. have the one library link to 2.x of the other).

**Comment by ikesyo** [comment]
It appears now that this is a duplicate of #803.
